src/main/java/com/multiplex/booking/BookingController.java

```diff
@@ -0,0 +1,279 @@
+    @GetMapping("/showingsByInterval/{dateTime}")
+    @ApiResponses(value = { 
+        @ApiResponse(responseCode = "200", description = "Found the showings in the time interval of +/- 2 hours", 
+            content = { @Content(mediaType = "application/json", 
+            schema = @Schema(implementation = Showing.class)) }),
+        @ApiResponse(responseCode = "400", description = "Invalid date supplied", 
+            content = @Content), 
+        @ApiResponse(responseCode = "404", description = "Showings not found", 
+            content = @Content) })
+    public List<Showing> getShowingsWithinTimeRange(@PathVariable("dateTime") LocalDateTime dateTime) {
+        LocalDateTime startTime = dateTime.minusHours(TIME_INTERVAL);
+        LocalDateTime endTime = dateTime.plusHours(TIME_INTERVAL);
+        return showings.findByShowingTimeBetween(startTime, endTime);
+    }
+
+    // user chooses a particular screening
+    @Operation(summary="Return a showing specified by id")
+    @GetMapping("/showingsById/{id}")
+    @ApiResponses(value = { 
+        @ApiResponse(responseCode = "200", description = "Found the showing", 
+            content = { @Content(mediaType = "application/json", 
+            schema = @Schema(implementation = Showing.class)) }),
+        @ApiResponse(responseCode = "400", description = "Invalid id supplied", 
+            content = @Content), 
+        @ApiResponse(responseCode = "404", description = "Showing not found", 
+            content = @Content) })
+    Showing one(@PathVariable Long id) {
+
+        return showings.findById(id).orElseThrow(() -> new ShowingNotFoundException(id));
+    }
+
+    @GetMapping("/seatByShowingIdRowColumn/{id}/{rowNr}/{columnNr}")
+    Seat one(@PathVariable Long id, @PathVariable int rowNr, @PathVariable int columnNr) {
+
+        Showing showing = showings.findById(id) .orElseThrow(() -> new ShowingNotFoundException(id));
+
+        Set<Seat> seats = showing.getSeats();
+        Seat foundSeat = new Seat();
+
+        for (Seat a : seats) {
+            if(a.getrowNr() == rowNr && a.getcolumnNr() == columnNr) foundSeat = a;
+        }
+
+        return foundSeat;
+    }
+
+    // user chooses seats, and gives the name of the person doing the reservation
+    // (name and surname)
+    @Operation(summary="Reservation of seat(s) by id. Id of seat, name, surname and ticket type is required.")
+    @PutMapping("/ReservationOfSeats")
+    @ApiResponses(value = { 
+        @ApiResponse(responseCode = "200", description = "Created new reservation", 
+            content = { @Content(mediaType = "application/json", 
+            schema = @Schema(implementation = Room.class)) }),
+        @ApiResponse(responseCode = "400", description = "Invalid json supplied", 
+            content = @Content)})
+    ReservationResponse replaceSeat(@Valid @RequestBody List<Seat> newSeats) {
+
+        Showing showing = showings.findById(newSeats.get(0).getShowingId())
```

Comment:
Co sie stanie jak podamy pusta listę? :)

Reply:
Dostalibyśmy prawdopodobnie błąd "IndexOutOfBoundsException", gdyż pusta lista nie ma żadnego elementu. Jednym z rozwiązań byłoby sprawdzenie czy lista nie jest pusta i jeśli jest - zwrócić błąd.
```suggestion
        if (newSeats.isEmpty()) {
             throw new WrongInputException();
        } else {
             Showing showing = showings.findById(newSeats.get(0).getShowingId());
        }
```